kernel/src/net/socket/unix/stream/connected.rs

```diff
@@ -1,65 +1,74 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use core::ops::Deref;
-
-use ostd::sync::PreemptDisabled;
-
 use crate::{
     events::IoEvents,
-    fs::utils::{Channel, Consumer, Producer},
+    fs::utils::{ChannelPollee, Peered},
     net::socket::{
         unix::{addr::UnixSocketAddrBound, UnixSocketAddr},
         util::SockShutdownCmd,
     },
     prelude::*,
-    process::signal::{PollHandle, Pollee},
-    util::{MultiRead, MultiWrite},
+    process::signal::Pollee,
+    util::{
+        ring_buffer::{RbConsumer, RbProducer, RingBuffer},
+        MultiRead, MultiWrite,
+    },
 };
 
 pub(super) struct Connected {
-    addr: AddrView,
-    reader: Consumer<u8>,
-    writer: Producer<u8>,
+    inner: Peered<Inner>,
+    reader: Mutex<RbConsumer<u8>>,
+    writer: Mutex<RbProducer<u8>>,
 }
 
 impl Connected {
     pub(super) fn new_pair(
         addr: Option<UnixSocketAddrBound>,
         peer_addr: Option<UnixSocketAddrBound>,
-        reader_pollee: Option<Pollee>,
-        writer_pollee: Option<Pollee>,
+        is_read_shutdown: bool,
+        is_write_shutdown: bool,
+        pollee: Pollee,
```

Comment:
How about directly passing two `ChannelPollee` here? Naming them as`this_pollee` and `peer_pollee` may convey more clear meanings.